Adds a few last minute bugs and advisory numbers (#48703)
openshift/openshift-docs: release_notes/ocp-4-11-release-notes.adoc

== About this release

// TODO: Update with the relevant information closer to release.

> {product-title} (link:https://access.redhat.com/errata/RHSA-2022:5069[RHSA-2022:5069]) is now available. This release uses link:https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.24.md[Kubernetes 1.24] with CRI-O runtime. New features, changes, and known issues that pertain to {product-title} {product-version} are included in this topic.

Author: That is correct. The link will work at GA.

* Previously, the registry used exact match when applying `ImageContentSourcePolicy`(ICSP). With this update, ICSP is applied for subrepositories and mirrors now work as expected. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2014240[*BZ#2014240*])

* Previously, the Image Registry Operator did not work with AWS S3 on {rh-openstack}. With this update, the Image Registry Operator trusts AWS S3 on all platforms. (link:https://bugzilla.redhat.com/show_bug.cgi?id=200761[*BZ#2007611*])

* Previously, {product-title} image registry did not work with Ceph Radosgw. With this update, the image registry works with Ceph Radosgw. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1976782[*BZ#1976782*])

> * Previously, the Image Registry Operator interpreted `429` error messages from upstream registries as if the data was not available. The Operator returned a `404 Not Found instead` message instead of `429 Too Many Requests`. With this update, the proper `429 Too Many Requests` message is returned so that administrators know to retry the request. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1923536[*BZ#1923536*])

Reviewer: Should "``404 Not Found instead`" just be "`404 Not Found`"?

* Previously, the Image Registry Operator could not be used to configure CloudFront. With this update, the Image Registry Operator can configure CloudFront. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2065224[*BZ#2065224*])

> * Previously, the Image Registry Operator push images when KMS encryption was enabled. With this update, images can be pushed and pulled with KMS encryption enabled. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2048214[*BZ#2048214*])

Reviewer: Should "Image Registry Operator push images" be "Image Registry Operator pushed images"?

> * Previously, the Image Registry Operator push images when KMS encryption was enabled. With this update, images can be pushed and pulled with KMS encryption enabled. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2048214[*BZ#2048214*])

Reviewer: From this line, I'm not certain what the change is between the previous behaviour and the updated behaviour. Was it that previously the IMO could not push images when KMS was enabled? If so, you could update this line to reflect that.

* With this update, {product-title} on {rh-virtualization-first} supports preallocated disks for control plane and worker nodes for installer-provisioned infrastructure. In high-load environments, preallocated disks benefit performance for etcd and other components. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2035334[*BZ#2035334*])

> * Previously, when you tried to delete multiple clusters in parallel, the process failed because of a bug in the `vmware/govmomi` library. The bug caused one of the cluster's tags to be deleted, which resulted in a `404` error when the process to delete could not find the tag. This update ignores tags that are not found and continues to delete so that it finishes without error. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2021041[*BZ#2021041*])

Reviewer 2: ```suggestion
* Previously, when you tried to delete multiple clusters in parallel, the process failed because of a bug in the `vmware/govmomi` library. The bug caused  a cluster tag to be deleted, which resulted in a `404` error when the process to delete could not find the tag. This update ignores tags that are not found and continues to delete so that it finishes without error. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2021041[*BZ#2021041*])
```

[id="ocp-4-11-machine-config-operator-bug-fixes"]
==== Machine Config Operator

> * Because `keyFile` is not configured as the default plug-in for NetworkManager on RHEL nodes, RHEL nodes might not reach the ready state after a reboot. With this fix, `keyFile` is set as the default NetworkManager plug-in on all cluster nodes. As a result, nodes correctly reach the ready state after a reboot. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2060133[*BZ#2060133*])

Reviewer 2: ```suggestion
* Because `keyFile` is not configured as the default plug-in for NetworkManager on RHEL nodes, RHEL nodes might not reach the ready state after a reboot. With this fix, `keyFile` is set as the default NetworkManager plug-in on all cluster nodes. As a result, nodes correctly reach the `ready` state after a reboot. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2060133[*BZ#2060133*])
```

[id="ocp-4-11-networking-bug-fixes"]
==== Networking

* Previously, when using the bond CNI plug-in for an additional network attachment, it was not compatible with Multus. When the bond CNI plug-in was used in conjunction with the Whereabouts IPAM plug-in for a network attachment definition, assigned IP addresses were incorrectly reconciled. Now a network attachment definition that uses the bond CNI plug-in works correctly with the Whereabouts IPAM plug-in for IP address assignment. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2082360[*BZ#2082360*])

> * Previously, when using the OVN-Kubernetes cluster network provider with multiple default gateways, the wrong gateway was selected, causing the OVN-Kubernetes pods to stop unexpectedly. Now the correct default gateway is selected such that these pods no longer crashes. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2040933[*BZ#2040933*])

Reviewer 2: ```suggestion
* Previously, when using the OVN-Kubernetes cluster network provider with multiple default gateways, the wrong gateway was selected, causing the OVN-Kubernetes pods to stop unexpectedly. Now the correct default gateway is selected such that these pods no longer fail. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2040933[*BZ#2040933*])
```

* Previously, a race condition in the `NetworkManager` restart sometimes prevented DHCP resolution to successfully complete setting up the `br-ex` bridge on node boot when using OVN-Kubernetes. With this update, `NetworkManager` is no longer restarted when setting up `br-ex`, eliminating the race condition. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2055433[*BZ#2055433*])

> * Previously, the `PtpConfigSlave` source custom resource (CR) was set to the unsupported network_transport UDPv4, causing errors on Distributed Unit (DU) nodes. This fix updates `PtpConfigSlave` source CR to use network_transport L2 rather than UDPv4. As a result, errors are no longer found on the DU nodes. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2060492#[*BZ#2060492*])

Reviewer: Should "network_transport" be "network transport"? My apologies if I didn't spot this in my first review. It is hard to know what has changed when a PR is updated.

Reviewer: You could update "This fix updates `PtpConfigSlave` source CR" to "This fix updates the `PtpConfigSlave` source CR".

* With this update, a `--subresource` flag was added to the `oc adm policy who-can` command to check who can perform a specified action on a subresource. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1905850[*BZ#1905850*])

> * Previously, users were unable to use tab completion for the `oc project` command. Now, hitting tab after `oc project` properly lists projects. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2080416[*BZ#2080416*])

Reviewer 2: ```suggestion
* Previously, users were unable to use tab completion for the `oc project` command. Now, hitting tab after `oc project` properly lists projects. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2080416[*BZ#2080416*])
```

* Previously, the `openshift-router` process briefly ignored the `SIGTERM` shutdown signal. This caused containers to ignore a Kubernetes shutdown request, resulting in one hour shutdown times. With this update, the router now responds to `SIGTERM` signals. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2076297#[*BZ#2076297*])

* Previously, when an admitted route’s Ingress Controller was deleted or a sharding configuration was added, a false status of `admitted` was given. With this update, the Ingress Controller clears the status of an `unadmitted` route avoiding the false status scenario. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1944851[*BZ#1944851*])

> * Previously, the Amazon Web Service (AWS) cloud provider implementation checks would mistakenly use an internal service-load-balance (SLB) to be configured due errors in the Ingress Operator reporting `Progressing=True` for the ClusterOperator status condition. This caused clusters upgrading to 4.10 not to complete. With this update, the Ingress Operator no longer prevents upgrades of clusters to complete. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2055470[*BZ#2055470*])

Reviewer 2: ```suggestion
* Previously, the Amazon Web Service (AWS) cloud provider implementation checks would mistakenly use an internal service-load-balance (SLB) to be configured due to errors in the Ingress Operator reporting `Progressing=True` for the ClusterOperator status condition. This caused clusters upgrading to 4.10 not to complete. With this update, the Ingress Operator no longer prevents upgrades of clusters to complete. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2055470[*BZ#2055470*])
```

* Previously, the `openshift-router` process briefly ignored the `SIGTERM` shutdown signal. This caused containers to ignore a Kubernetes shutdown request, resulting in one hour shutdown times. With this update, the router now responds to `SIGTERM` signals. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2076297#[*BZ#2076297*])

* Previously, when an Ingress Controller for an admitted route was deleted or a sharding configuration was added, a false status of `admitted` was given. With this update, the Ingress Controller clears the status of an `unadmitted` route, avoiding the false status scenario. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1944851[*BZ#1944851*])

> * Previously, {product-title} clusters installed using version 4.7 or earlier maintained a service.beta.kubernetes.io/aws-load-balancer-internal annotation value of 0.0.0.0/0. Clusters that were installed using 4.8 or later have the annotation value true. AWS cloud-provider implementations that check for the annotation value true would return the wrong result if the value was 0.0.0.0/0. As a result, cluster upgrades to 4.10 would not complete. With this update, the annotation value is normalized to true so that the cluster upgrades can complete. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2055470[*BZ#2055470*])

Reviewer: In my suggested text for this line, I included back-ticks but I think they were rendered in the comment rather than being printed explicitly. As such, a copy and paste from that comment might have not captured the back-ticks. Please add back-ticks for each instance of the following elements in this line:

* `service.beta.kubernetes.io/aws-load-balancer-internal`
* `0.0.0.0/0`
* `true`

* With this update, the default credentials request for Amazon Web Services (AWS) is modified to allow mounting of encrypted volumes using customer managed keys from KMS. Administrators who created credentials requests in manual mode with CCO will need to apply those changes manually if they intend to mount encrypted volumes using customer managed keys on AWS. Other administrators should not be impacted by this change. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2049872[*BZ#2049872*])

> * Previously, after destroying the cluster, the IBM Volume remained un-deleted, preventing the cluster from being completely destroyed. This fix adds support in the installation program and CSI driver, resulting in the volume’s deletion after the cluster is destroyed. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2047732#[*BZ2047732*])

Reviewer 2: ```suggestion
* Previously, after deleting the cluster, the IBM Volume remained un-deleted, preventing the cluster from being completely deleted. This fix adds support in the installation program and CSI driver, resulting in the volume’s deletion after the cluster is removed. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2047732#[*BZ2047732*])
```

* With this update, the `jsonData` property has been deprecated and replaced with `data` in the `ping` source. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2084438[*BZ#2084438*])

* Previously, topology with more than 100 nodes was failing or lagging due to size. With this update, users are alerted to the lagging issue and offered the option to use `Search` to find a specific resource or continue with topology eliminating any lag. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2060329[*BZ#2060329*])

* Previously, The `8080` port was stalled on the service port list and random ports are deleted on a service with multiple ports. With this update, a condition in the `createService` function is specified to identify the current route `tartgetPort` and replace that prot from the service ports whenever a new `targetPort` is added in the edit application form. Now, when port 8080 is replaced as route port, it is removed from the service ports list and the new route ports list and the new route port gets added to the list. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2077943[*BZ#2077943*])

> * Previously, the git detection used to manage instance APIs to get repository information did not work for repositories from self hosted GitHub and Bitbucket. With this update, detection for self hosted GitHub and Bitbucket instance repositories works. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2038244[*BZ#2038244*])

Reviewer: You could place "git" in back-ticks if it refers to the `git` command. Otherwise, you could capitalise the word if it refers to the product name.

> * Previously, `apiVersion` was not passed in the correct format to `Resource` dropdown in `EventSource` creation form. This caused `InContext` to not be selected under `EventSource` creation, which excluded it from the `Resource` dropdown menu. With this update, the `Resource` dropdown menu includes `Resource` from `InContext`. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2070020[*BZ#2070020*])

Reviewer: Are `Resource` and `EventSource` UI elements in a form? If so, you could update them to use bold format instead of using back-ticks. Back-ticks typically refer to code elements.

Also, you could change "**Resource** dropdown" to "the **Resource** drop-down menu", to follow the ISG guidelines. This applies to each of the references to that menu in this line.

> * Previously, the *pipeline metrics* page displayed all API calls for the metrics query and are failing with a `404` error. With this update, the `prometheus-tenancy` API to get the metrics data for the pipeline. Now, the pipeline metrics page displays all data and graphs to the non-admin user with at least view access to the namespace. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2041769[*BZ#2041769*])

Reviewer: Should "Pipeline" be capitalised, when referring to the "Pipeline metrics" page?

> * Previously, the *pipeline metrics* page displayed all API calls for the metrics query and are failing with a `404` error. With this update, the `prometheus-tenancy` API to get the metrics data for the pipeline. Now, the pipeline metrics page displays all data and graphs to the non-admin user with at least view access to the namespace. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2041769[*BZ#2041769*])

Reviewer: You could change "are failing" to "failed", to use the past tense consistently in the first line.

> * Previously, the *pipeline metrics* page displayed all API calls for the metrics query and are failing with a `404` error. With this update, the `prometheus-tenancy` API to get the metrics data for the pipeline. Now, the pipeline metrics page displays all data and graphs to the non-admin user with at least view access to the namespace. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2041769[*BZ#2041769*])

Reviewer: Should "the `prometheus-tenancy` API to get the metrics" be "the `prometheus-tenancy` API is used to get the metrics "?

> * Previously, you could access a quick search and add modal with the Crtl+space keyboard shortcut, but you could not close them by the same keyboard shortcut. With this update, you can now close the quick search and add modal using Ctrl+space keyboard shortcut. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2093586[*BZ#2093586*])

Reviewer: You could update "close them by the same keyboard shortcut" to "close them by using the same keyboard shortcut".

> * Previously, you could access a quick search and add modal with the Crtl+space keyboard shortcut, but you could not close them by the same keyboard shortcut. With this update, you can now close the quick search and add modal using Ctrl+space keyboard shortcut. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2093586[*BZ#2093586*])

Reviewer: You could change "you can now close" to "you can close" in the second sentence.

Also in the second sentence, you could update "and add modal using Ctrl+space keyboard shortcut" to "and add modal using the Ctrl+space keyboard shortcut".

> * Previously,  the resources created for user settings are not removed if the user is deleted. Consequently, the created resources in the `openshift-console-user-settings` namespace is never removed. With this update, an `ownerReference` is added to your metadata at creation. This allows automatic removal of the resources when the user no longer exists. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2019564[*BZ#2019564*])

Reviewer: You could update "are not removed if the user is deleted" to "were not removed if the user was deleted" to use past tense to describe the behaviour before the fix.

Reviewer: The same applies to "namespace is never removed", you could update it to "namespace were never removed".

* Before this update, some of the sidebar items did not contain namespace context. Consequently, when links were opened from another browser, or opening links from a different active namespace, the web console does not switch to the correct namespace. With this update, the correct namespace is selected when opening the URL. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2039647[*BZ#2039647*])

> * Previously, when using the console to instantiate a template, its parameters were stored as a secret resource. When the template was removed, the secret remained. This consequently created an unnecessary buildup of secrets in a cluster. With this update, an ownership reference is added to the secret that maps to the template instance. Now, when the template instance is removed, the secret is also removed. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2015042[*BZ#2015042*])

Reviewer: Should "buildup" be "build up"? My apologies if I missed this before.

* With this update, the `jsonData` property has been deprecated and replaced with `data` in the `ping` source. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2084438[*BZ#2084438*])

> * Previously, the topology view in the {product-title} web console would fail or lag for clusters with more than 100 nodes. With this update, the topology view shows a `LimitExceeded` state for clusters with more than 100 nodes. An option is provided to view the resources by using the *Search* page instead. Alternatively, you can click Show topology anyway to continue to load the topology view. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2060329[*BZ#2060329*])

Reviewer: Please use bold font for "Show topology anyway", which is a UI element.

* Previously, if a service exposed multiple service ports and the route target port was `8080`, attempts to change the target port would result in another service port being updated instead of the port `8080` service port. With this update, the service port corresponding to the active target port is replaced when a new target port is set. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2077943[*BZ#2077943*])

* Previously, the `git` detection used to manage instance APIs to get repository information did not work for repositories from self hosted GitHub and Bitbucket. With this update, detection for self hosted GitHub and Bitbucket instance repositories works. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2038244[*BZ#2038244*])

> * Previously, `apiVersion` was not passed in the correct format to *Resource* drop-down menu in the *EventSource* creation form. This caused `InContext` to not be selected under *EventSource* creation, which excluded it from the *Resource* drop-down menu. With this update, the *Resource* drop-down menu includes `Resource` from `InContext`. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2070020[*BZ#2070020*])

Reviewer: You could update "the correct format to *Resource* drop-down menu" to "the correct format to the *Resource* drop-down menu".

* Previously, the *Pipeline metrics* page displayed all API calls for the metrics query and failed with a `404` error. With this update, the `prometheus-tenancy` API is used to get the metrics data for the pipeline. Now, the pipeline metrics page displays all data and graphs to the non-admin user with at least view access to the namespace. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2041769[*BZ#2041769*])

* Previously, you could access a quick search and add modal with the Crtl+space keyboard shortcut, but you could not close them by using the same keyboard shortcut. With this update, you can close the quick search and add modal using the Ctrl+space keyboard shortcut. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2093586[*BZ#2093586*])

> * Previously,  the resources created for user settings were not removed if the user is deleted. Consequently, the created resources in the `openshift-console-user-settings` namespace was never removed. With this update, an `ownerReference` is added to your metadata at creation. This allows automatic removal of the resources when the user no longer exists. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2019564[*BZ#2019564*])

Reviewer: You could change "if the user is deleted" to "if the user was deleted" in the first line.

+
(link:https://issues.redhat.com/browse/KATA-1373[*KATA-1373*])

> * There is currently a known issue that degrades the Machine Config Operator (MCO) when you attempt to create a cluster with both FIPS and `realTimeKernel`. As a workaround, modify the merge logic, which will prevent the MCO from degrading. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2096496[*BZ#2096496*])

Reviewer: I am unsure if the second sentence in this line is a recommendation to the reader or something that is being considered as a fix for this known issue. This might leave the reader wondering if they are supposed to do something to work around this. You could update this line to provide clarity about this. If it is something for the reader to do, you could provide an additional resources reference.

> * The pipeline metrics API does not support the required `pipelinerun/taskrun` histogram values from OSP 1.6 and beyond. Consequently, the *Metrics* tab in the *Pipeline* -> *Details* page is removed instead of displaying incorrect values. There is currently no workaround for this issue. (link: https://bugzilla.redhat.com/show_bug.cgi?id=2074767[*BZ#2074767*])

Reviewer: Does OSP refer to OpenStack Platform? If so, you could replace "OSP" with "{rh-openstack}".
